File: Excel/602 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/602 Clarks Triangle.xlsx"
test  = read_excel(path, range = "A2:U12", col_names = F) %>% as.matrix() %>% replace(is.na(.), 0)

create_clarks_matrix = function(n) {
  mat = matrix(0, nrow = n, ncol = n * 2 - 1)
  mid = n
  for (i in 1:n) {
    for (j in 1:(n * 2 - 1)) {
      if (j == mid - (i - 1)) {
        mat[i, j] = ifelse(i == 1, 0, 1)
      } else if (j == mid + (i - 1)) {
        mat[i, j] = 6 * (i - 1)
      } else if (i > 1 && j > 1 && j < n * 2 - 1 && mat[i - 1, j - 1] != 0 && mat[i - 1, j + 1] != 0) {
        mat[i, j] = mat[i - 1, j - 1] + mat[i - 1, j + 1]
      }
    }
  }
  return(mat)
}

n = 11
clark = create_clarks_matrix(n)

all.equal(clark, test, check.attributes = F)
# [1] TRUE                                                                          

File: Excel/600-699/602/602 Challenge.py
import pandas as pd
import numpy as np

path = "602 Clarks Triangle.xlsx"
test = pd.read_excel(path,header=None, usecols="A:U", skiprows=1, nrows=11).fillna(0).to_numpy()

def create_clarks_matrix(n):
    mat = np.zeros((n, n * 2 - 1), dtype=int)
    mid = n - 1
    for i in range(n):
        for j in range(n * 2 - 1):
            if j == mid - i:
                mat[i, j] = 0 if i == 0 else 1
            elif j == mid + i:
                mat[i, j] = 6 * i
            elif i > 0 and j > 0 and j < n * 2 - 2 and mat[i - 1, j - 1] != 0 and mat[i - 1, j + 1] != 0:
                mat[i, j] = mat[i - 1, j - 1] + mat[i - 1, j + 1]
    return mat

n = 11
clark = create_clarks_matrix(n)

print(np.array_equal(clark, test))
# True